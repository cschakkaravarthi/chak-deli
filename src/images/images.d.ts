declare const images: {
    bannerSrc: any;
    logoSrc: any;
    siteLogo: any;
    noNotification: any;
    artistBg: any;
    brandBg: any;
    systemBg: any;
    defaultAvatar: any;
    placeholder: any;
    add: any;
    faceBrowser: any;
    faceDocument: any;
    faceHeart: any;
    faceLock: any;
    faceStop: any;
    chromeIcon: any;
    safariIcon: any;
    edgeIcon: any;
    external: any;
    allLinksGrid: any;
    gearIcon: any;
    defaultIcon: any;
    remove: any;
    defaultIconCircle: any;
    spinner: any;
};
export default images;
//# sourceMappingURL=images.d.ts.map