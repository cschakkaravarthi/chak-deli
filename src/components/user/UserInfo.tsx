import React, { FC } from 'react';
import get from 'lodash.get';
import UserInfoModel, { Phone } from '../../shared/models/UserInfo.model';

type Props = {
  usersInfo?: UserInfoModel;
};

const UserInfo: FC<Props> = props => {
  const { usersInfo } = props;

  if (!usersInfo || !usersInfo.networkID) return null;

  return (
    <>
      <div className='user__info'>
        <div className='user__name'>
          <a href='#'>
            {usersInfo.lastName}, {usersInfo.firstName}
          </a>
        </div>
        <div className='user__title'>
          <p>{usersInfo.jobTitle}</p>
          <p>{usersInfo.department}</p>
        </div>
        <div className='user__email'>
          <a type='email' href={`mailto:${usersInfo.email}`}>
            {usersInfo.email}
          </a>
        </div>
        {get(usersInfo, 'location.office') && (
          <div className='user__office'>
            <p>Office: {get(usersInfo, 'location.office', '')}</p>
          </div>
        )}
        <div className='user__address'>
          <p>
            {get(usersInfo, 'location.room', '')} <br />
            {get(usersInfo, 'location.address', '')} <br />
            {`${get(usersInfo, 'location.city', '')}, ${get(usersInfo, 'location.state', '')} `}
            {get(usersInfo, 'location.postalCode', '')}
          </p>
        </div>
        <div className='user__phone'>
          {get(usersInfo, 'phone', [] as Phone[]).map((item: Phone, index: number) => (
            <p key={index}>
              {item.type === 'Desk' && <span>Phone: {item.num}</span>}
              {item.type === 'Mobile' && <span>Mobile: {item.num}</span>}
            </p>
          ))}
        </div>
      </div>
    </>
  );
};

export default UserInfo;
