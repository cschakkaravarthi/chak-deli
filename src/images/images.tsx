const images = {
  bannerSrc: require('../assets/banner.jpg'),
  logoSrc: require('../assets/logo.png'),
  siteLogo: require('../assets/umgc-logo.svg'),
  noNotification: require('../assets/no_notif.svg'),
  artistBg: require('../assets/artists.jpg'),
  brandBg: require('../assets/brands_and_labels.jpg'),
  systemBg: require('../assets/systems_and_applications.jpg'),
  defaultAvatar: require('../assets/avatar.svg'),
  placeholder: require('../assets/placeholder.jpg'),
  add: require('../assets/add.svg'),
  faceBrowser: require('../assets/face_browser.svg'),
  faceDocument: require('../assets/face_document.svg'),
  faceHeart: require('../assets/face_heart.svg'),
  faceLock: require('../assets/face_lock.svg'),
  faceStop: require('../assets/face_stop.svg'),
  chromeIcon: require('../assets/Chrome-Logo-Icon.png'),
  safariIcon: require('../assets/Safari_logo.png'),
  edgeIcon: require('../assets/Microsoft-Edge.png'),
  external: require('../assets/external.svg'),
  allLinksGrid: require('../assets/all_links_grid.svg'),
  gearIcon: require('../assets/gear.svg'),
  defaultIcon: require('../assets/default_icon.svg'),
  remove: require('../assets/remove.svg'),
  defaultIconCircle: require('../assets/default_icon_circle.svg'),
  spinner: require('../assets/spinner.gif'),
  caretRight: require('../assets/caret-right.svg'),
  caretLeft: require('../assets/caret-left.svg'),
  appreciationCardSampleTop: require('../assets/ap-text.svg'),
  appreciationCardSampleBottom: require('../assets/man-guitar.svg'),
  removeFilter: require('../assets/times-circle.svg')
};

export default images;
