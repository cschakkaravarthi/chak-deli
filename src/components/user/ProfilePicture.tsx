import React, { FC } from 'react';
import { UserPhoto } from '../../shared/models/UserInfo.model';

type Props = {
  profilePictureSrc?: UserPhoto | any;
};

const ProfilePicture: FC<Props> = props => {
  const { profilePictureSrc } = props;

  return (
    <div className='user__picture'>
      <img src={profilePictureSrc} alt='user profile picture' />
    </div>
  );
};

export default ProfilePicture;
