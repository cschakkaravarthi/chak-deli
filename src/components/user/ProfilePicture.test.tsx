import React from 'react';
import TestRenderer from 'react-test-renderer';
import ProfilePicture from './ProfilePicture';

import get from 'lodash.get';

import { dummyUserInfo } from '../../shared/models/UserInfo.model';
import images from '../../images/images';

describe('ProfilePicture', () => {
  it('renders correctly with info', () => {
    const tree = TestRenderer.create(
      <ProfilePicture
        profilePictureSrc={get(
          dummyUserInfo,
          'photo.medium',
          images.defaultAvatar
        )}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
