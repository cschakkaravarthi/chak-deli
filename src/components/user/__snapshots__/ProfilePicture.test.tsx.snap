// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePicture renders correctly with info 1`] = `
<div
  className="user__picture"
>
  <img
    alt="user profile picture"
    src="test-file-stub"
  />
</div>
`;
